app/src/main/java/com/iti/android/team1/ebuy/ui/category/view/CategoryFragment.kt

```diff
@@ -173,7 +175,7 @@ class CategoryFragment : Fragment() {
         )
         binding.productRecycler.apply {
             this.adapter = categoryProductsAdapter
-            this.layoutManager = GridLayoutManager(context, 2)
+            this.layoutManager = StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL)
```

Comment:
what is difference between them

Reply:
StaggeredGridLayoutManager automatically sets the item views according to its height and width.
so when the recycler view gets inflated you will get this view in return

![Screenshot (130)](https://user-images.githubusercontent.com/60023467/171754029-c2a8da15-4303-449e-be99-71f494e136bc.png)

Unlike GridLayoutManager the height and width of each grid are equal, ending up with this view 
![Screenshot (131)](https://user-images.githubusercontent.com/60023467/171754084-0363877b-d1e3-47c8-b152-ad9269003b6e.png)

Reply:
you can read more from this link: https://www.geeksforgeeks.org/recyclerview-as-staggered-grid-in-android-with-example/